homeassistant/components/switch/rest.py

```diff
@@ -75,32 +81,47 @@ def is_on(self):
 
     def turn_on(self, **kwargs):
         """Turn the device on."""
+        body_on_t = template.render_with_possible_json_value(
+            self._hass, self._body_on, "", False)
         request = requests.post(self._resource,
-                                data=self._body_on,
-                                timeout=10)
+                                data=body_on_t,
+                                timeout=50)
         if request.status_code == 200:
             self._state = True
         else:
-            _LOGGER.error("Can't turn on %s. Is resource/endpoint offline?",
-                          self._resource)
+            _LOGGER.info("Can't turn on %s. Is resource/endpoint offline?",
+                         -                          self._resource)
 
     def turn_off(self, **kwargs):
         """Turn the device off."""
+        body_off_t = template.render_with_possible_json_value(
```

Comment:
I assume you meant `self._body_off` in the comment about using `template.render`. 
`self._value_template` is supposed to compute whether the state of the switch is on, from the response to the GET request.
Also, I don't understand what you meant by "only render if there is a value_template".